lib/active_model/serializer.rb

```diff
@@ -199,16 +199,22 @@ def serializer_from_options(options)
       opts
     end
 
+    def self.serializers_cache
+      @serializers_cache ||= ThreadSafe::Cache.new
```

Comment:
I'm not very familiar with this. Can you explain what this does and whether there are any potential side effects?